packages/react-router/lib/hooks.tsx

```diff
@@ -399,3 +397,18 @@ export function _renderMatches(
     );
   }, null as React.ReactElement | null);
 }
+
+// Note: OutletProps/Outlet are included in thr hooks file so they can be leveraged
+// by renderMatches without creating a circular dependency
+export interface OutletProps {
+  context?: unknown;
+}
+
+/**
+ * Renders the child route's element, if there is one.
+ *
+ * @see https://reactrouter.com/docs/en/v6/api#outlet
+ */
+export function Outlet(props: OutletProps): React.ReactElement | null {
+  return useOutlet(props.context);
+}
```

Comment:
This was the reason for the circular dependency:

* `components.tsx` needs lots of things from `hooks.tsx`
* But then `useRoutes` calls `_renderMatches` which needed `Outlet` from `components.tsx`

So by moving `Outlet` into `hooks.tsx` we avoid having to import from `components.tsx`.  If this feels wrong we can go back to storing components and hooks in the same file?

Reply:
Actually @mjackson this open PR might allow us to move `Outlet` back into `components.tsx`...

https://github.com/remix-run/react-router/pull/8497/files#diff-b60f1a2d4276b2a605c05e19816634111de2e8a4186fe9dd7de8e344b65ed4d3R1073

Reply:
That PR looks great. You wanna go ahead and merge it and then rebase on top of it?

Reply:
Confirmed circular dependency is resolved with that PR